src/main/java/dekk/pw/pokemate/tasks/EvolvePokemon.java

```diff
@@ -33,7 +34,8 @@
     @Override
     public void run(Context context) {
         try {
-            for (Pokemon pokemon : context.getApi().getInventories().getPokebank().getPokemons()) {
+            List<Pokemon> pokemonList = context.getApi().getInventories().getPokebank().getPokemons();
```

Comment:
This literally change nothing, replace this line with:
`new ArrayList<>(context.getApi().getInventories().getPokebank().getPokemons());`.

That would actually fix the issue as it clones the list.

Remember to import ArrayList.